LoginClient/PipeClient.cs

```diff
@@ -0,0 +1,368 @@
+        static extern SafeFileHandle CreateFile(
+           String pipeName,
+           uint dwDesiredAccess,
+           uint dwShareMode,
+           IntPtr lpSecurityAttributes,
+           uint dwCreationDisposition,
+           uint dwFlagsAndAttributes,
+           IntPtr hTemplate);
+
+        /// <summary>
+        /// Handles the messages recieved from a server pipe.
+        /// </summary>
+        /// <param name="message">The byte message recieved</param>
+        public delegate void MessageRecievedHandler(byte[] message);
+
+        /// <summary>
+        /// The event that is called whenever a message is recieved from the server pipe.
+        /// </summary>
+        public event MessageRecievedHandler MessageRecieved;
+
+        /// <summary>
+        /// Handles the server disconnected message.
+        /// </summary>
+        public delegate void ServerDisconnectedHandler();
+
+        /// <summary>
+        /// The Event that is called when the server pipe is severed.
+        /// </summary>
+        public event ServerDisconnectedHandler ServerDisconnected;
+
+        /// <summary>
+        /// Handles the client connected to the server message.
+        /// </summary>
+        public delegate void ConnectedToServerHandler();
+
+        /// <summary>
+        /// The event that is called when the client connects to the server.
+        /// </summary>
+        public event ConnectedToServerHandler ConnectedToServer;
+
+        const int BUFFER_SIZE = 4096;
+
+        FileStream stream;
+        SafeFileHandle handle;
+        Thread readThread;
+
+        /// <summary>
+        /// Is the client connected to a server pipe.
+        /// </summary>
+        private bool connected;
+
+        /// <summary>
+        /// The name of the pipe connected to the server.
+        /// </summary>
+        private string pipeName;
+
+        /// <summary>
+        /// Is the client logged in as the admin.
+        /// </summary>
+        private bool isLoggedIn;
+
+        /// <summary>
+        /// The methods for accessing the 
+        /// connected, pipename, and isLoggedOn properties.
+        /// </summary>
+        /// <returns></returns>
+        #region Accessors
+
+        public bool IsLoggedIn()
+        {
+            return isLoggedIn;
+        }
+
+        private void IsLoggedIn(bool isLoggedInOrNot)
+        {
+            isLoggedIn = isLoggedInOrNot;
+
+            if (isLoggedIn)
+            {
+                ConnectedToServer();
+            }
+        }
+
+        /// <summary>
+        /// Returns true if client is connected to a server.
+        /// </summary>
+        /// <returns>Boolean</returns>
+        public bool IsConnected()
+        {
+            return connected;
+        }
+
+        /// <summary>
+        /// Set the value of the connected boolean.
+        /// </summary>
+        /// <param name="b">The new value of connected.</param>
+        private void IsConnected(bool b)
```

Comment:
Better to use  public bool IsConnected{ get; private set; } for properties like this.
so its clearer that this doesnt set anything (functions mean actions are being taken.)
But this is just a C# standard thing, not a requirement.